### PR TITLE
W2 - D3 - BaoLe - Android DrawerLayout

### PR DESCRIPTION
**Pull request by at-baole**
W2 - D3 - BaoLe - Android DrawerLayout #18

### CODE REVIEW

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import android.media.MediaScannerConnection;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.util.Log'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import java.util.List;
+import java.util.Objects;
+
+import asiantech.internship.summer.R;
+import asiantech.internship.summer.drawerlayout.model.Item;
+
+
+public class DrawerLayoutActivity extends AppCompatActivity implements RecyclerAdapter.OnItemListener {
+    //private static final String TAG = DrawerLayoutActivity.class.getSimpleName();
+    private int GALLERY = 1;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [SuspiciousConstantFieldName] The field name indicates a constant but its modifiers do not
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#SuspiciousConstantFieldName

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    private int CAMERA = 2;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [SuspiciousConstantFieldName] The field name indicates a constant but its modifiers do not
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#SuspiciousConstantFieldName

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/RecyclerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @Override
+    public int getItemCount() {
+        return itemObjects.size();
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        if (isPositionHeader(position))
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/RecyclerAdapter.java]: WARNING: [IfStmtsMustUseBraces] Avoid using if statements without curly braces
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/braces.html#IfStmtsMustUseBraces

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+import java.util.List;
+import java.util.Objects;
+
+import asiantech.internship.summer.R;
+import asiantech.internship.summer.drawerlayout.model.Item;
+
+
+public class DrawerLayoutActivity extends AppCompatActivity implements RecyclerAdapter.OnItemListener {
+    //private static final String TAG = DrawerLayoutActivity.class.getSimpleName();
+    private static int GALLERY = 1;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [SuspiciousConstantFieldName] The field name indicates a constant but its modifiers do not
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#SuspiciousConstantFieldName

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    private static int CAMERA = 2;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [SuspiciousConstantFieldName] The field name indicates a constant but its modifiers do not
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#SuspiciousConstantFieldName

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 7]
```diff
@@ -0,0 +1,177 @@
+package asiantech.internship.summer.drawerlayout;
+
+import android.Manifest;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.Manifest'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.karumi.dexter.Dexter;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.karumi.dexter.Dexter'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+import com.karumi.dexter.MultiplePermissionsReport;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.karumi.dexter.MultiplePermissionsReport'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+import com.karumi.dexter.PermissionToken;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.karumi.dexter.PermissionToken'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+import com.karumi.dexter.listener.PermissionRequest;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.karumi.dexter.listener.PermissionRequest'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+import com.karumi.dexter.listener.multi.MultiplePermissionsListener;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.karumi.dexter.listener.multi.MultiplePermissionsListener'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/RecyclerAdapter.java [hunk 13]
```diff
@@ -0,0 +1,79 @@
+package asiantech.internship.summer.drawerlayout;
+
+import android.content.Context;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/RecyclerAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.content.Context'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/RecyclerAdapter.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/RecyclerAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.widget.Toast'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/model/DrawerItem.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
+package asiantech.internship.summer.drawerlayout.model;
+
+import android.graphics.Bitmap;
+
+public class DrawerItem {
+    private int icon;
+    private String title;
+    private boolean isSelected;
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/model/DrawerItem.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isSelected has the same name as a method
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.List;
+
+import asiantech.internship.summer.R;
+import asiantech.internship.summer.drawerlayout.model.DrawerItem;
+
+public class DrawerAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    //private static final String TAG = DrawerAdapter.class.getSimpleName();
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: Xoá dòng này

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+    private static final int TYPE_HEADER = 0;
+    private static final int TYPE_ITEM = 1;
+    private List<DrawerItem> mListItems;
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: Đổi tên thành mItems hoặc mItemList

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+            View layoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout, parent, false);
+            return new ItemViewHolder(layoutView);
+        }
+        throw new RuntimeException();
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+        DrawerItem mItem = mListItems.get(position);
+        if (holder instanceof HeaderViewHolder) {
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: Tạo 1 hàm onBind trong ViewHolder và gọi nó tại đây

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+            if (mItem.getAvatarBitmap() != null) {
+                ((HeaderViewHolder) holder).mImgAvatar.setImageBitmap(mItem.getAvatarBitmap());
+            } else {
+                ((HeaderViewHolder) holder).mImgAvatar.setImageResource(mItem.getItemImage());
+
+            }
+            ((HeaderViewHolder) holder).mTvEmail.setText(mItem.getItemText());
+            ((HeaderViewHolder) holder).mImgAvatar.setOnClickListener(view -> mListener.onAvatarClicked());
+        } else if (holder instanceof ItemViewHolder) {
+            ((ItemViewHolder) holder).mImgItem.setImageResource(mItem.getItemImage());
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: Tạo 1 hàm onBind trong ViewHolder và gọi nó tại đây

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+            ((ItemViewHolder) holder).mTvItem.setText(mItem.getItemText());
+            ((ItemViewHolder) holder).mTvItem.setTextColor(mContext.getResources().getColorStateList(R.color.bg_item_color));
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: Dùng itemView.getContext() thay cho context

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 21]
```diff
@@ -0,0 +1,208 @@
...
+import java.util.Calendar;
+import java.util.List;
+import java.util.Objects;
+
+import asiantech.internship.summer.R;
+import asiantech.internship.summer.drawerlayout.model.DrawerItem;
+
+
+public class DrawerLayoutActivity extends AppCompatActivity implements DrawerAdapter.OnItemClickListener {
+    //private static final String TAG = DrawerLayoutActivity.class.getSimpleName();
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Xoá dòng này

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 22]
```diff
@@ -0,0 +1,208 @@
...
+    private static final int GALLERY = 1;
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Những hằng số như thế này nên định nghĩa số lớn để tránh trùng nhau

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 23]
```diff
@@ -0,0 +1,208 @@
...
+    private List<DrawerItem> mItems;
+    private DrawerAdapter mAdapter;
+    private int mPositionSelected = -1;
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_drawer_layout);
+        //requestMultiplePermission();
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Xoá comment code

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 24]
```diff
@@ -0,0 +1,208 @@
...
+        initView();
+        initDrawer();
+    }
+
+    private void initView() {
+        mockData();
+        RecyclerView recyclerView;
+        recyclerView = findViewById(R.id.recyclerViewHeader);
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Ghép 2 dòng 54, 55 lại thành 1

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 25]
```diff
@@ -0,0 +1,208 @@
...
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(DrawerLayoutActivity.this);
+        recyclerView.setLayoutManager(linearLayoutManager);
+        recyclerView.setHasFixedSize(true);
+        mAdapter = new DrawerAdapter(mItems, this, getApplicationContext());
+        recyclerView.setAdapter(mAdapter);
+    }
+
+    public void initDrawer() {
+        DrawerLayout mDrawerLayout = findViewById(R.id.drawerLayout);
+        TextView mTvName = findViewById(R.id.tvName);
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: findViewById nên để trong cùng 1 hàm

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 26]
```diff
@@ -0,0 +1,208 @@
...
+    }
+
+    @Override
+    public void onItemClicked(int position) {
+        if (mPositionSelected != -1) {
+            mItems.get(mPositionSelected).setIsSelected(false);
+        }
+        mPositionSelected = position;
+        mItems.get(position).setIsSelected(true);
+        mAdapter.notifyDataSetChanged();
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Dùng notifyItemChange

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 27]
```diff
@@ -0,0 +1,208 @@
...
+        String pictureDIalogItems[] = {
+                getString(R.string.selectGallery),
+                getString(R.string.captureCamera)
+        };
+        pictureDialog.setItems(pictureDIalogItems, (dialogInterface, selectAction) -> {
+            switch (selectAction) {
+                case 0:
+                    choosePhotosFromGallery();
+                    break;
+                case 1:
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Case 0 là gì, 1  là gì

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 28]
```diff
@@ -0,0 +1,208 @@
...
+            FileOutputStream fo = new FileOutputStream(f);
+            fo.write(bytes.toByteArray());
+            MediaScannerConnection.scanFile(this, new String[]{f.getPath()}, new String[]{getString(R.string.imagePathPNG)}, null);
+            fo.close();
+            f.getAbsolutePath();
+        } catch (IOException ignored) {
+        }
+    }
+
+//    private void requestMultiplePermission() {
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Xoá những code comment nếu không dùng

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/HeaderViewHolder.java [hunk 29]
```diff
@@ -0,0 +1,19 @@
+package asiantech.internship.summer.drawerlayout;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import asiantech.internship.summer.R;
+
+class HeaderViewHolder extends RecyclerView.ViewHolder {
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/HeaderViewHolder.java]: Move vào trong adapter

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/model/DrawerItem.java [hunk 30]
```diff
@@ -0,0 +1,40 @@
+package asiantech.internship.summer.drawerlayout.model;
+
+import android.graphics.Bitmap;
+
+public class DrawerItem {
+    private int icon;
+    private String title;
+    private boolean checkSelected;
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/model/DrawerItem.java]: Check convention cách đặt tên biến kiểu boolean

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 31]
```diff
@@ -0,0 +1,87 @@
...
+public class DrawerAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    //private static final String TAG = DrawerAdapter.class.getSimpleName();
+    private static final int TYPE_HEADER = 0;
+    private static final int TYPE_ITEM = 1;
+    private List<DrawerItem> mListItems;
+    private OnItemClickListener mListener;
+    private Context mContext;
+
+    DrawerAdapter(List<DrawerItem> listItems, OnItemClickListener listener, Context context) {
+        this.mListItems = listItems;
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: Không cần dùng this.

File: app/src/at-baole/res/layout/item_layout.xml [hunk 32]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-huongnguyen) [inline app/src/at-baole/res/layout/item_layout.xml]: Đổi tên class thành drawer_item

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 33]
```diff
@@ -0,0 +1,114 @@
...
+        void onAvatarClicked();
+
+        void onItemClicked(int position);
+    }
+
+    class HeaderViewHolder extends RecyclerView.ViewHolder {
+        private ImageView mImgAvatar;
+        private TextView mTvEmail;
+
+        HeaderViewHolder(View itemView, OnItemClickListener headerListener) {
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: Ko cần headerListener, dùng mListener

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 34]
```diff
@@ -0,0 +1,114 @@
...
+            mTvEmail.setText(header.getItemText());
+        }
+    }
+
+    class ItemViewHolder extends RecyclerView.ViewHolder {
+        private TextView mTvItem;
+        private LinearLayout mllItem;
+        private ImageView mImgItem;
+
+        ItemViewHolder(View itemView, OnItemClickListener itemListener) {
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: Dùng mListener

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java [hunk 35]
```diff
@@ -0,0 +1,114 @@
...
+            mTvItem = itemView.findViewById(R.id.tvItem);
+            mllItem = itemView.findViewById(R.id.llItem);
+            itemView.setOnClickListener(view -> itemListener.onItemClicked(getLayoutPosition()));
+        }
+
+        void onBindItemView(DrawerItem item) {
+            mImgItem.setImageResource(item.getItemImage());
+            mTvItem.setText(item.getItemText());
+            mllItem.setSelected(item.isSelected());
+            mTvItem.setTextColor(itemView.getContext().getResources().getColorStateList(R.color.bg_item_color));
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerAdapter.java]: TextColor set trong xml file

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 36]
```diff
@@ -0,0 +1,179 @@
...
+                    break;
+                case CAPTURE_CAMERA:
+                    capturePhotosFromCamera();
+                    break;
+            }
+        });
+        pictureDialog.show();
+    }
+
+    public void choosePhotosFromGallery() {
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Make private và check permission runtime

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 37]
```diff
@@ -0,0 +1,179 @@
...
+        Intent galleryIntent = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+        startActivityForResult(galleryIntent, GALLERY);
+    }
+
+    public void capturePhotosFromCamera() {
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Make private và check permission runtime

File: app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java [hunk 38]
```diff
@@ -0,0 +1,179 @@
...
+        } else if (requestCode == CAMERA) {
+            Bitmap cameraBitmap = (Bitmap) Objects.requireNonNull(data.getExtras()).get(getString(R.string.data));
+            item.setAvatarBitmap(cameraBitmap);
+            saveImage(cameraBitmap);
+            mAdapter.notifyItemChanged(0);
+            Toast.makeText(DrawerLayoutActivity.this, R.string.imageSaved, Toast.LENGTH_LONG).show();
+        }
+    }
+
+    public void saveImage(Bitmap myBitmap) {
```
Comment (at-huongnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/drawerlayout/DrawerLayoutActivity.java]: Make private

**Pull request merged**